### PR TITLE
Simplified install with PyPI `rdkit` and git install in `setup.py`

### PR DESCRIPTION
**Pull request by JacksonBurns**
## Description
The current install instructions require the user to separately install RDKit from the appropriate conda channel, and then install `descriptastorus` with `pip` separately before then installing `chemprop`. Now that [RDKit is on PyPI](https://pypi.org/project/rdkit-pypi/), the entire install can be consolidated into just `pip install chemprop` by adding the two dependencies to the `install_requires`.
Creating a fresh environment and running `pip install chemprop; pytest -v` passes all unit tests on my end.

## Bugfix / Desired workflow
This was all that added to `setup.py`:
```
        'rdkit>=2020.03.1.0',
        'descriptastorus @ git+https://github.com/bp-kelley/descriptastorus'
```

## Questions
 - README might need to be updated more than what I changed

## Checklist
- [x] linted with flake8?
~- [ ] (if appropriate) unit tests added?~

### CODE REVIEW

**Comment by davidegraff** [comment]
looks like the build is failing for some reason with the pip install?

**Comment by JacksonBurns** [comment]
build is failing during the conda install step, which this PR should not affect. It's also failing [on master](https://github.com/chemprop/chemprop/actions/runs/3964928049/jobs/6794209590) after the latest commit.

I removed the lines from the CI workflow that try to install the dependencies from conda (I think that's what I did, anyway) so now it should rely only on the pip install, just as a check. Let's see if it runs?

**Comment by davidegraff** [comment]
fair point- at the point that you're updating the packaging, can you:
1. move things over to use `setup.cfg` where possible
2. update the README (as it looks like we no longer need `conda` to set up the environment)
3. update the actions as well (see (2)- I don't think we need to rely on setting up the testing environment via `conda`

**Comment by JacksonBurns** [comment]
> fair point- at the point that you're updating the packaging, can you:
> 
> 1. move things over to use `setup.cfg` where possible

pypa is planning to [deprecate `setup.cfg`](https://github.com/pypa/setuptools/issues/3214) in the relatively near future, maybe the install should be left in `setup.py` or moved to a `pyproject.toml` now?

> 2. update the README (as it looks like we no longer need `conda` to set up the environment)

It's probably still best practice to recommend that users have conda up and running to ensure that they (1) have the correct version of python and (2) don't attempt to install `chemprop` in their base python environment, which would inevitably cause package conflicts if they want to run anything else.

> 3. update the actions as well (see (2)- I don't think we need to rely on setting up the testing environment via `conda`

I did this on another branch on my fork [here](https://github.com/JacksonBurns/chemprop/actions/runs/4026398325/jobs/6920878653) and will add the changes here as well. I think you will still need to manually allow the new workflow.
This does not work on MacOS because the bash option `-l` is being used - I don't know why this is here or really what it does for `chemprop` but can it be removed? [Here is an issue](https://github.com/actions/setup-python/issues/179#issuecomment-750495666) on the `setup-python` action explaining why this fails.

> n.b. descriptastorus can't be added as a package dependency when uploading to PyPI as it's not on PyPI. You'll have to make a note in the README to tell users to install it separately.

I don't know what n.b. means, but as you can see in the workflow I linked above the install works with just pip. `descriptastorus` is not on PyPI, but the `pip install git+...` trick can just be done in the `install_requires` section, as I have done [here](https://github.com/chemprop/chemprop/pull/364/commits/9deada5908a5a62cf0fba2c028c320d883f64759#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R54).

**Comment by davidegraff** [comment]
> pypa is planning to [deprecate `setup.cfg`](https://github.com/pypa/setuptools/issues/3214) in the relatively near future, maybe the install should be left in `setup.py` or moved to a `pyproject.toml` now?

I’m ambivalent to moving towards `pyproject.toml` if you want to do that but
1. The depreciation  is still in discussion, so even if it’s adopted the move is still years away
2. The move will be accompanied by a tool to translate a `setup.cfg` to the corresponding `pyproject.toml`, not `setup.py`
3. The general point still remains that we should move towards a packaging system that separates the build ”logic” (`setup.py`) from the “configuration”. How we do that isn’t of too much concern to me

> It's probably still best practice to recommend that users have conda up and running to ensure that they (1) have the correct version of python and (2) don't attempt to install `chemprop` in their base python environment, which would inevitably cause package conflicts if they want to run anything else.

1. it’s possible to do that without telling users to install via the environment.yml file.
2. It’s not really up to us to tell users the environment isolation system they should be using if we can avoid it (because it’s now possible to avoid `conda` entirely)
3. You can specify allowable python versions inside `setup.cfg`

> I did this on another branch on my fork [here](https://github.com/JacksonBurns/chemprop/actions/runs/4026398325/jobs/6920878653) and will add the changes here as well. I think you will still need to manually allow the new workflow. This does not work on MacOS because the bash option `-l` is being used - I don't know why this is here or really what it does for `chemprop` but can it be removed? [Here is an issue](https://github.com/actions/setup-python/issues/179#issuecomment-750495666) on the `setup-python` action explaining why this fails.

This flag is needed for when environments in GH actions rely on `conda`. It should be removable with a new non-`conda` based environment

> I don't know what n.b. means, but as you can see in the workflow I linked above the install works with just pip. `descriptastorus` is not on PyPI, but the `pip install git+...` trick can just be done in the `install_requires` section, as I have done [here](https://github.com/chemprop/chemprop/pull/364/commits/9deada5908a5a62cf0fba2c028c320d883f64759#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R54).

You can’t upload a package that lists a git dependency in install_requires to PyPI. See [1](https://stackoverflow.com/a/54894359), [2](https://github.com/python-poetry/poetry/issues/2828)

**Comment by JacksonBurns** [comment]
> I’m ambivalent to moving towards `pyproject.toml` if you want to do that but
> 
> 1. The depreciation  is still in discussion, so even if it’s adopted the move is still years away
> 2. The move will be accompanied by a tool to translate a `setup.cfg` to the corresponding `pyproject.toml`, not `setup.py`
> 3. The general point still remains that we should move towards a packaging system that separates the build ”logic” (`setup.py`) from the “configuration”. How we do that isn’t of too much concern to me
> 1. it’s possible to do that without telling users to install via the environment.yml file.
> 2. It’s not really up to us to tell users the environment isolation system they should be using if we can avoid it (because it’s now possible to avoid `conda` entirely)
> 3. You can specify allowable python versions inside `setup.cfg`

All fair points - I will update the README and port everything from `setup.py` to `setup.cfg` with [`setuptools-py2cfg`](https://github.com/gvalkov/setuptools-py2cfg). IT doesn't catch entry scripts, but I don't know how to write those in the ini format either.

> This flag is needed for when environments in GH actions rely on `conda`. It should be removable with a new non-`conda` based environment

Removed it, [runs fine now](https://github.com/JacksonBurns/chemprop/actions/runs/4028774701).

> You can’t upload a package that lists a git dependency in install_requires to PyPI. See [1](https://stackoverflow.com/a/54894359), [2](https://github.com/python-poetry/poetry/issues/2828)

Turns out that `descriptastorus` actually _is_ on [PyPI](https://pypi.org/project/descriptastorus/). I searched on the repo and it turns out the [maintainers did it a while ago](https://github.com/bp-kelley/descriptastorus/pull/19) but never updated the README. I changed the install requirement to match.

**Comment by davidegraff** [comment]
oh nice thanks for finding PyPI package for `descriptastorus`. All of this looks good, just waiting to make sure the actions complete without a problem here (I see they've already worked for you.) I'll approve this after that's done.

**Review by davidegraff — APPROVED** [review]
siiiick 👍

**Comment by oscarwumit** [comment]
Thanks for the contribution. I will go ahead and merge.

**Pull request merged**
